### DISCUSSION TITLE
Scheduled launch

### DISCUSSION THREAD

**Discussion by andyvauliln** [FlowiseAI/Flowise#63] [Ideas]
Good to have ability to run chat flows with defined schedule

**Comment by HenryHengZJ** [comment]
Mind to clarify your use case? Currently Flowise is used mostly for chat application, not yet for automated task

**Reply by ufxpri** [reply]
It can be utilized for a wide range of tasks. For instance, it can provide personalized restaurant recommendations based on the user's current location and time. Additionally, it can send periodic emotion status check messages to help monitor the user's mental well-being. Furthermore, it is capable of composing daily logs or messages for the user, allowing for efficient communication and record-keeping.

**Reply by HenryHengZJ** [reply]
that makes sense! Will definitely add the cron job ability down the line when we expand Flowise beyond just the chat functionality

**Reply by Webmasterei** [reply]
I like to add, that we would need a method to add content incrementally to pinecone. (Maybe by scheduled crawls?) 
We like to use it with a wiki that is constantly expanding and updating.

**Comment by ufxpri** [comment]
This function needs to be classified as "trigger" since other applications could have this kind of action.
As an example we can add message trigger, iot trigger, ext.

**Comment by Lotykun** [comment]
This Idea is created on May.. how about this?... thanks

**Comment by chattymaps** [comment]
Following this for updates.  A scheduler would be great.

**Comment by AdamDudley** [comment]
You could add this as an additional Start node.

Allow a table of Chron and Start Prompt, similar to the State table
![image](https://github.com/user-attachments/assets/eec25bfa-d5c9-4549-bb6f-7e665ca5e809)
(ignore my first  row)

Then just initiate the 'chat'/'sequence with that prompt.
Please allow us to add a variable or update the current state with the *current data and time*

Examples of use.

1) for ISO27001 we need to remind people about recent security breaches to stay vigilant; a sequence to find the latest news in month <todays month> <todays year>, and if that doesn't find any thing, then fall back to other interesting items of new related to our programming languages and security.

2) send a weekly summary of xxxxx to myself

3) re-read the database that's being populated with related reddit posts and fill in the corresponding data field with results of the chain.

Some of these use cases slide over into what you can do with ActivePieces, n8n and similar. Where as flowise is more a chatbot creator, so you may wish to be clear this isn't a primary use case for the app, but simple to add with nice functionality. All can be done today though the APIs, so you're adding a shortcut :)
